sdk/search/search-documents/test/public/node/searchClient.spec.ts

```diff
@@ -254,7 +254,10 @@ describe("SearchClient", function(this: Suite) {
     assert.equal(documentCount, 11);
   });
 
-  it("search with speller", async function() {
+  // Fails in CI because the CI search service was created on or before 2019
+  // which does not have search 'speller' feature. Will resolve the
+  // resource issue and then add this test back.
+  it.skip("search with speller", async function() {
```

Comment:
Is this because the search service created by CI is using an older API version? 

If so, this is fine to fix the build but consider creating an issue and adding support for multi-version-testing like we did in [key vault](https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/keyvault/keyvault-keys/test/public/crypto.hsm.spec.ts#L16) - added benefit of ensuring previous service versions are compatible and allows you to skip live tests if the service API version is older than X

Feel free to reach out for a tour on how we did in Key Vault

Reply:
That is great suggestion @maorleger I will look into the keyvault and create an issue. Will sync up with you on implementing it. Thanks